### ISSUE TITLE
02. PR1: GlobalAnchorCoarseDataset の output に file_index / primary_key_id / secondary_key_id を追加する

### ISSUE THREAD

**Issue by ozw4** [ozw4/seisai#421]
# 02. PR1: GlobalAnchorCoarseDataset の output に file_index / primary_key_id / secondary_key_id を追加する

## 目的

`GlobalAnchorCoarseDataset` の各 sample に、DataLoader でそのまま batch tensor として collate できる identity field を追加する。

後続 PR の trace-memory consistency loss は、文字列 field ではなく tensor field を使って current view の source primary/secondary を判定する。

## 依存 issue

- `01_issue_define_trace_identity_helpers.md`

## 対象ファイル

```text
packages/seisai-engine/src/seisai_engine/pipelines/fbpick/coarse/build_dataset.py
packages/seisai-engine/tests/test_fbpick_coarse.py  # test は issue 04 で追加してもよい
```

## 追加する output contract

`GlobalAnchorCoarseDataset._build_global_anchor_sample()` が返す `out` に、以下の top-level field を追加する。

```text
file_index:       scalar torch.Tensor, dtype=torch.int64
primary_key_id:   scalar torch.Tensor, dtype=torch.int64
secondary_key_id: scalar torch.Tensor, dtype=torch.int64
```

例:

```python
out['file_index']       # tensor(0, dtype=torch.int64)
out['primary_key_id']   # tensor(0, dtype=torch.int64)  # ffid
out['secondary_key_id'] # tensor(1, dtype=torch.int64)  # chno
```

### 1. `_build_global_anchor_sample()` 内で id を計算する

`sample` と `info` が利用可能になった後、`build_output_base()` を呼ぶ前に以下を計算する。

```python
file_index = self._file_index_for_info(info)
primary_key_id = _primary_key_id(sample['key_name'])
secondary_key_id = _secondary_key_id(sample['secondary_key'])
```

### 2. `build_output_base(..., extra={...})` に追加する

既存の `extra` に以下を追加する。

```python
'file_index': torch.tensor(file_index, dtype=torch.int64),
'primary_key_id': torch.tensor(primary_key_id, dtype=torch.int64),
'secondary_key_id': torch.tensor(secondary_key_id, dtype=torch.int64),
```

既存の文字列 field は消さない。

```text
out['key_name']       # 既存 contract。維持する。
out['secondary_key']  # 既存 contract。維持する。
out['primary_unique'] # 既存 contract。維持する。
```

### 3. `meta` には原則追加しない

この PR の目的は collate しやすい tensor field を top-level に追加すること。`meta['key_name']` や `meta['secondary_key']` など既存の人間可読情報は維持するが、今回追加する id は top-level output とする。

後続 PR の loss は `batch['file_index']`, `batch['primary_key_id']`, `batch['secondary_key_id']` を参照する前提にする。

## 非目標

この issue では以下を実装しない。

- `anchor_raw_indices_view` の追加
- key id を config 化すること
- `GlobalAnchorCoarseRawInferDataset` の output 変更
- memory bank / consistency loss 実装

## 受け入れ条件

- `GlobalAnchorCoarseDataset` の sample top-level に `file_index`, `primary_key_id`, `secondary_key_id` が存在する。
- 3 field はすべて scalar `torch.Tensor` で、dtype は `torch.int64`。
- `file_index` は `self.file_infos` における 0-based index と一致する。
- `primary_key_id.item()` は `out['key_name']` の mapping と一致する。
- `secondary_key_id.item()` は `out['secondary_key']` の mapping と一致する。
- 既存の `key_name`, `secondary_key`, `primary_unique`, `indices`, `anchor_raw_indices` は維持される。

## 確認コマンド

```bash
pytest packages/seisai-engine/tests/test_fbpick_coarse.py -q
```

issue 04 の test をまだ入れていない場合でも、既存 test を壊していないことを確認する。

## 実装上の注意

`torch.tensor(np.int64_value)` でも動くが、dtype は明示する。

```python
torch.tensor(int(file_index), dtype=torch.int64)
```

DataLoader の default collate では scalar tensor が `(B,)` になる。後続 loss はその形を想定する。
